sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/PassThroughThenCleanup.java

```diff
@@ -79,6 +81,6 @@ public void processElement(ProcessContext c) {
   }
 
   abstract static class CleanupOperation implements Serializable {
-    abstract void cleanup(PipelineOptions options) throws Exception;
+    abstract void cleanup(DoFn.ProcessContext options) throws Exception;
```

Comment:
Something like this?

class JobIdContainer {

public JobIdContainer(DoFn<?, ?>.ProcessContext pc, PCView<> view) {
this.view = ...
this.pc = ...
}

public getJobId() {
return …
}

}